CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationRewards.uc

```diff
@@ -47,30 +52,78 @@ static protected function bool RewardNotAvaliable(optional XComGameState NewGame
 //////////////////////
 // Granted when the mission is completed
 
-static function X2RewardTemplate CreateGatherLeadDarkEventReward()
+static function X2RewardTemplate CreateGatherLeadActivityReward()
+{
+	local X2RewardTemplate Template;
+	
+	`CREATE_X2Reward_TEMPLATE(Template, 'Reward_GatherLeadActivity');
+	Template.GetRewardStringFn = GatherLeadString;
+	Template.GiveRewardFn = GiveGatherLeadActivity;
+
+	return Template;
+}
+
+static function X2RewardTemplate CreateGatherLeadTargetReward()
+{
+	local X2RewardTemplate Template;
+	
+	`CREATE_X2Reward_TEMPLATE(Template, 'Reward_GatherLeadTarget');
+	Template.GetRewardStringFn = GatherLeadString;
+	Template.GiveRewardFn = GiveGatherLeadTarget;
+
+	return Template;
+}
+
+static function X2RewardTemplate CreateGatherLeadPersonnelReward()
 {
 	local X2RewardTemplate Template;
 	
-	`CREATE_X2Reward_TEMPLATE(Template, 'Reward_GatherLeadDarkEvent');
+	`CREATE_X2Reward_TEMPLATE(Template, 'Reward_GatherLeadPersonnel');
 	Template.GetRewardStringFn = GatherLeadString;
-	Template.GiveRewardFn = GiveGatherLeadDarkEvent;
+	Template.GiveRewardFn = GiveGatherLeadPersonnel;
 
 	return Template;
 }
 
 static protected function string GatherLeadString(XComGameState_Reward RewardState)
 {
-	return "Gather lead";
+	return "Gather Lead";
 }
 
-static protected function GiveGatherLeadDarkEvent(XComGameState NewGameState, XComGameState_Reward RewardState, optional StateObjectReference AuxRef, optional bool bOrder=false, optional int OrderHours=-1)
+static protected function GiveGatherLeadActivity(XComGameState NewGameState, XComGameState_Reward RewardState, optional StateObjectReference AuxRef, optional bool bOrder=false, optional int OrderHours=-1)
 {
 	local XComGameState_ResistanceFaction FactionState;
 
 	FactionState = XComGameState_ResistanceFaction(NewGameState.ModifyStateObject(class'XComGameState_ResistanceFaction', FindRandomMetFaction().ObjectID));
 
 	SpawnCovertAction(NewGameState, FactionState, 'CovertAction_P2DarkEvent');
 	SpawnCovertAction(NewGameState, FactionState, 'CovertAction_P2DarkEvent');
+
+	class'UIUtilities_Infiltration'.static.InfiltrationActionAvaliable(AuxRef, NewGameState);
+}
+
+static protected function GiveGatherLeadTarget(XComGameState NewGameState, XComGameState_Reward RewardState, optional StateObjectReference AuxRef, optional bool bOrder=false, optional int OrderHours=-1)
+{
+	local XComGameState_ResistanceFaction FactionState;
+
+	FactionState = XComGameState_ResistanceFaction(NewGameState.ModifyStateObject(class'XComGameState_ResistanceFaction', FindRandomMetFaction().ObjectID));
+
+	SpawnCovertAction(NewGameState, FactionState, 'CovertAction_P2SupplyRaid');
+	SpawnCovertAction(NewGameState, FactionState, 'CovertAction_P2DarkVIP');
+
+	class'UIUtilities_Infiltration'.static.InfiltrationActionAvaliable(AuxRef, NewGameState);
```

Comment:
The first argument is the action to focus when opening the screen. In this case `AuxRef.ObjectID` is always `0` (you can check the mission source code)

I suggest that you make the first argument in `InfiltrationActionAvaliable` optional and just omit it here